src/routers/sessions.py

```
@@ -460,12 +462,24 @@ async def get_session_context(
-    query: str = Body(..., description="Search query"),
+    search: schemas.MessageSearchOptions = Body(
+        ..., description="Message search parameters "
+    ),
     db: AsyncSession = db,
 ):
+    query, use_semantic_search = search.query, search.use_semantic_search
+    embed_messages_enabled = settings.LLM.EMBED_MESSAGES
+    if use_semantic_search and not embed_messages_enabled:
+        raise DisabledException(
+            "Semantic search requires EMBED_MESSAGES flag to be enabled"
+        )
```

Ruff (0.11.9)

Avoid specifying long messages outside the exception class

(TRY003)